src/org/traccar/protocol/ArnaviBinaryProtocolDecoder.java

```diff
@@ -0,0 +1,273 @@
+package org.traccar.protocol;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.DeviceSession;
+import org.traccar.helper.Log;
+import org.traccar.model.Position;
+
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ArnaviBinaryProtocolDecoder extends BaseProtocolDecoder {
+
+    public ArnaviBinaryProtocolDecoder(ArnaviBinaryProtocol protocol) {
+        super(protocol);
+    }
+
+    private static final int PROTOCOL_VERSION1 = 0x22;
+    private static final int PROTOCOL_VERSION2 = 0x23;
+
+    private static final byte PACKET_DATA = 0x01;
+    private static final byte PACKET_TEXT = 0x03;
+    private static final byte PACKET_FILE = 0x04;
+    private static final byte PACKET_DATA_BINARY = 0x06;
+
+    private static final byte TAG_VOLTAGE_SUPPLY = 1;
+    private static final byte TAG_ID_KEY = 2;
+    private static final byte TAG_LATITUDE = 3;
+    private static final byte TAG_LONGITUDE = 4;
+    private static final byte TAG_COORD_PARAMS = 5;
+    private static final byte TAG_DEVICE_STATUS = 9;
+    private static final byte TAG_FREQ_0 = 20;
+    private static final byte TAG_FREQ_1 = 21;
+    private static final byte TAG_FREQ_2 = 22;
+    private static final byte TAG_FREQ_3 = 23;
+    private static final byte TAG_FREQ_4 = 24;
+    private static final byte TAG_FREQ_5 = 25;
+    private static final byte TAG_FREQ_6 = 26;
+    private static final byte TAG_FREQ_7 = 27;
+    private static final byte TAG_VOLT_0 = 30;
+    private static final byte TAG_VOLT_1 = 31;
+    private static final byte TAG_VOLT_2 = 32;
+    private static final byte TAG_VOLT_3 = 33;
+    private static final byte TAG_VOLT_4 = 34;
+    private static final byte TAG_VOLT_5 = 35;
+    private static final byte TAG_VOLT_6 = 36;
+    private static final byte TAG_VOLT_7 = 37;
+    private static final byte TAG_LLS_0 = 70;
+    private static final byte TAG_LLS_1 = 71;
+    private static final byte TAG_LLS_2 = 72;
+    private static final byte TAG_LLS_3 = 73;
+    private static final byte TAG_LLS_4 = 74;
+    private static final byte TAG_LLS_5 = 75;
+    private static final byte TAG_LLS_6 = 76;
+    private static final byte TAG_LLS_7 = 77;
+    private static final byte TAG_LLS_8 = 78;
+    private static final byte TAG_LLS_9 = 79;
+
+
+    private List<Position> positions;
+
+
+    private static int crc8(byte[] data) {
```

Comment:
All CRC calculations should be in `Checksum` class. It's likely that this one is already implemented there.